lib/javascript.d.ts

```diff
@@ -341,17 +346,17 @@ declare global {
 			 * this can be called synchronously and immediately returns the state.
 			 * Otherwise you need to provide a callback.
 			 */
-			getState: (callback?: GetStateCallback) => void | State;
+			getState: <T = any>(callback?: GetStateCallback<T>) => void | State<T> | AbsentState;
 
 			/**
 			 * Sets all queried states to the given value.
 			 */
-			setState: (id: string, state: string | number | boolean | State | Partial<State>, ack?: boolean, callback?: SetStateCallback) => this;
+			setState: <T extends string | number | boolean>(id: string, state: T | State<T> | Partial<State<T>>, ack?: boolean, callback?: SetStateCallback) => this;
```

Comment:
I guess this is more elegant this way, so ok.

Reply:
But let's add `Record<string, any>` for object-type states and `(string | number | boolean)[]` for array-type states.
We should probably extract that constraint into its own type, e.g. `StateValue` and use it to constrain the Parameter in `State` and `ChangedStateObject`.